### PR TITLE
Include the path to the configuration value in conversion error messages

### PR DESCRIPTION
**Pull request by mvestergaard**
It is currently very difficult to figure out which value causes errors when using configuration binding.

This change will include the path to the value in error messages.

Addresses #779

### CODE REVIEW

**Comment by mvestergaard** [comment]
I'm not sure why CI is failing. It doesn't seem related to my changes.

File: src/Configuration/Config.Binder/test/ConfigurationBinderTests.cs [hunk 1]
```diff
@@ -387,13 +387,13 @@ public void ConsistentExceptionOnFailedBinding(Type type)
             Assert.NotNull(exception.InnerException);
             Assert.NotNull(getException.InnerException);
             Assert.Equal(
-                Resources.FormatError_FailedBinding(IncorrectValue, type),
+                Resources.FormatError_FailedBinding(IncorrectValue, "Value", type),
```
Comment (HaoK) [inline src/Configuration/Config.Binder/test/ConfigurationBinderTests.cs]: Can you add a test that demonstrates what this change actually would look like with a path?

Comment (mvestergaard) [reply src/Configuration/Config.Binder/test/ConfigurationBinderTests.cs]: Doesn't it already demonstrate it with this test?
Using an in memory collection as the configuration provider, it's kind of forced to be a flat key/value map, so adding `A:More:Elaborate:Path` doesn't really prove anything. It's basically the same thing.
But if you disagree, I can add it.

Comment (mvestergaard) [reply src/Configuration/Config.Binder/test/ConfigurationBinderTests.cs]: Or are you looking for an example of binding to a deeper object graph?

Comment (HaoK) [reply src/Configuration/Config.Binder/test/ConfigurationBinderTests.cs]: Yes, a deeper path test that demonstrates how this would be useful

Comment (mvestergaard) [reply src/Configuration/Config.Binder/test/ConfigurationBinderTests.cs]: Test added

**Comment by Tratcher** [comment]
Fixed the CI with https://github.com/aspnet/Extensions/pull/1082, please rebase.

**Comment by mvestergaard** [comment]
Done

**Comment by Eilon** [comment]
@HaoK / @ajcvickers - is this ready to merge?

**Comment by ajcvickers** [comment]
@HaoK Can you review this an merge if you think it makes sense. (Note I don't think the path can contain sensitive data, so including it should be fine. The value itself should not be in the exception message, since it could be sensitive.)

**Comment by HaoK** [comment]
Had to rebase, merged as part of https://github.com/aspnet/Extensions/pull/1819  Thanks @mvestergaard !

**Pull request closed without merging**